frontend/src/components/AutoProduceConsumeComponent.js

```diff
@@ -0,0 +1,200 @@
+
+class AutoProduceConsume extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isCheckedConsume: false,
+      isCheckedProduce: false,
+      consumeRate: 1,
+      produceRate: 1,
+      time: 0,
+    };
+  }
+
+  componentDidMount() {
+    this.timerID = setInterval(this.count, 2000);
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.timerID);
+  }
+
+  handleConsumeSwitchChange = isCheckedConsume => this.setState({ isCheckedConsume });
+
+  handleConsumeRateChange = consumeRate => this.setState({ consumeRate });
+
+  handleProduceSwitchChange = isCheckedProduce => this.setState({ isCheckedProduce });
+
+  handleProduceRateChange = produceRate => this.setState({ produceRate });
+
+  pinPastTasks = (consumedTime) => {
+    const body = {
+      'problem-fact-change': {
+        $class: 'TaPinTaskProblemFactChange',
+        consumedTime,
+      },
+    };
+    const successMsg = `Pinned tasks that start before ${consumedTime} successfully.`;
+
+    submitProblemFactChange(
+      body, successMsg, this.props.container.containerId, this.props.solver.id,
+    );
+  };
+
+  randomInt = n => Math.floor(Math.random() * n);
+
+  pickRandomItem = arr => arr[this.randomInt(arr.length)];
+
+  addRandomTask = () => {
+    const readyTime = this.state.time;
+    const priority = this.pickRandomItem(constants.PRIORITIES);
+    const taskTypeId = this.pickRandomItem(this.props.taskTypes).id;
+    const customerId = this.pickRandomItem(this.props.customers).id;
+    const body = {
+      'problem-fact-change': {
+        $class: 'TaAddTaskProblemFactChange',
+        task: {
+          readyTime,
+          priority,
+          pinned: false,
+        },
+        taskTypeId,
+        customerId,
+      },
+    };
+    const successMsg = `Added task ${JSON.stringify(body['problem-fact-change'])}`;
+    submitProblemFactChange(
+      body, successMsg, this.props.container.containerId, this.props.solver.id,
+    );
+  }
+
+  count = () => {
+    isEveryProplemFactChangeProcessed(this.props.container.containerId, this.props.solver.id)
+      .then((isProcessed) => {
+        if (!isProcessed) {
+          console.log('Waiting until every problemFactChange is processed');
+          return;
+        }
+        if (this.state.isCheckedConsume) {
+          this.setState(
+            prevState => (
+              { time: prevState.time + (constants.MINUTE_STEP * prevState.consumeRate) }
+            ),
+            () => {
+              this.pinPastTasks(this.state.time);
+            },
+          );
+        }
+
+        if (this.state.isCheckedProduce) {
+          for (let i = 0; i < this.state.produceRate; i += 1) {
+            this.addRandomTask();
+          }
+        }
+
+        if (this.state.isCheckedConsume || this.state.isCheckedProduce) {
+          // Wait until problmFactChanges are processed
+          setTimeout(this.props.updateBestSolution, 500);
```

Comment:
How do you know that the PFC will be processed within 500ms? :slightly_smiling_face:

Reply:
The only way to know is by calling isEveryProblemFactProcessed (or REST call /problemFactChanges/processed), but even if not the UI should at least update based on what's currently processed. That's why I only introduced a short delay to give the server some time to process some/all PFCs.
Since it's not possible right now to communicate from the server to the UI that all PFCs are processed, I don't see a way to ensure that before updating the UI. What do you think?